physical/raft/raft_autopilot.go

```diff
@@ -803,3 +825,131 @@ func (b *RaftBackend) SetupAutopilot(ctx context.Context, storageConfig *Autopil
 
 	go b.startFollowerHeartbeatTracker()
 }
+
+// Ensure that the CustomPromoter implements the autopilot.Promoter interface
+var _ autopilot.Promoter = (*CustomPromoter)(nil)
+
+type CustomPromoter struct{}
+
+func (_ *CustomPromoter) GetServerExt(_ *autopilot.Config, srv *autopilot.ServerState) interface{} {
+	return nil
+}
+
+func (_ *CustomPromoter) GetStateExt(_ *autopilot.Config, _ *autopilot.State) interface{} {
+	return nil
+}
+
+func (_ *CustomPromoter) GetNodeTypes(c *autopilot.Config, s *autopilot.State) map[raft.ServerID]autopilot.NodeType {
+	types := make(map[raft.ServerID]autopilot.NodeType)
+	nonVoters := c.Ext.(map[raft.ServerID]bool)
+	for id := range s.Servers {
+		// If the server is a non-voter, mark it as such
+		if _, ok := nonVoters[id]; ok {
+			types[id] = autopilot.NodeType("non-voter")
+		} else {
+			types[id] = autopilot.NodeVoter
+		}
+	}
+	return types
+}
+
+func (_ *CustomPromoter) FilterFailedServerRemovals(_ *autopilot.Config, _ *autopilot.State, failed *autopilot.FailedServers) *autopilot.FailedServers {
+	return failed
+}
+
+// CalculatePromotionsAndDemotions will return a list of all promotions and demotions to be done as well as the server id of
+// the desired leader. This particular interface implementation maintains a stable leader and will promote healthy servers
+// to voting status if they are not makred as permanent non-voters. It will never change the leader ID nor will it perform demotions.
```

Comment:
```suggestion
// to voting status if they are not marked as permanent non-voters. It will never change the leader ID nor will it perform demotions.
```